### PR TITLE
Bug Fix - Editor renders multiple images #5

### PR DESCRIPTION
**Pull request by rajitha1998**
Fixes - #5 
## Changes proposed in the pull request
  * Fixing the multiple image rendering bug in mat-editor 
## Impact
 * Mat-editor library
 * Mat-editor demo
 * package.json (`node-sass` was upgraded)

### CODE REVIEW

**Review by psnmissaka — APPROVED** [review]
LGTM!

File: projects/demo/src/app/app.component.ts [hunk 1]
```diff
@@ -7,22 +7,38 @@ import { Component, OnInit } from '@angular/core';
...
-    document.getElementById("mat-editor").style.marginLeft= "10%";
+  public imageURL: string;
+
+  ngOnInit() {
+    document.getElementById("mat-editor").style.marginRight = "10%";
+    document.getElementById("mat-editor").style.marginLeft = "10%";
+  }
+
+  public uploadImage(event): void {
+    // This fuction can be used to upload images through the Angular app, which has integrated with the mat-editor
```
Comment (Anuranie) [inline projects/demo/src/app/app.component.ts]: which is integrated with the Mat-Editor

File: projects/demo/src/app/app.component.ts [hunk 2]
```diff
@@ -7,22 +7,38 @@ import { Component, OnInit } from '@angular/core';
...
+
+    /* Capture the image from the event, upload it to a object storage container and then set the Image url of the
```
Comment (Anuranie) [inline projects/demo/src/app/app.component.ts]: upload it to an object

File: projects/mat-editor/src/lib/common/execute-commands.service.ts [hunk 3]
```diff
@@ -191,9 +189,30 @@ export class ExecuteCommandsService {
       }
     }
   }
+
   /**
-   * 
+   * adds a loading tag using a html paragraph element
```
Comment (Anuranie) [inline projects/mat-editor/src/lib/common/execute-commands.service.ts]: Adds a loading tag using an HTML paragraph element

File: projects/mat-editor/src/lib/common/execute-commands.service.ts [hunk 4]
```diff
@@ -191,9 +189,30 @@ export class ExecuteCommandsService {
...
+      const restored = Utils.restoreSelection(this.savedSelection);
+      let html = '<br><p id="matLoading">' + message + '</p>';
+      if (restored) {
+        const inserted = document.execCommand('insertHTML', false, html);
+      }
+    }
+  }
+
+  /**
+  *  removes an html element in the editor content using the html id
```
Comment (Anuranie) [inline projects/mat-editor/src/lib/common/execute-commands.service.ts]: Removes an HTML element the editor's content area using the HTML ID

File: projects/mat-editor/src/lib/mat-editor-toolbar/mat-editor-toolbar.component.ts [hunk 5]
```diff
@@ -62,14 +62,15 @@ export class MatEditorToolbarComponent implements OnInit {
 
   ngOnInit() {
     this.imageURLSubscription = this.imageUploadService.afterImageUpload.subscribe((url) => {
+      // subscribing for a image url which is sent from the mat-editor component
```
Comment (Anuranie) [inline projects/mat-editor/src/lib/mat-editor-toolbar/mat-editor-toolbar.component.ts]: // Subscribing to an image URL which is sent from the Mat-Editor component

File: projects/mat-editor/src/lib/mat-editor-toolbar/mat-editor-toolbar.component.ts [hunk 6]
```diff
@@ -62,14 +62,15 @@ export class MatEditorToolbarComponent implements OnInit {
...
       this.imageURL = url;
       try {
+        //removing the loader after image gets added to the content
```
Comment (Anuranie) [inline projects/mat-editor/src/lib/mat-editor-toolbar/mat-editor-toolbar.component.ts]: // Removing the....

File: projects/mat-editor/src/lib/mat-editor.component.ts [hunk 7]
```diff
@@ -19,10 +19,16 @@ import * as Utils from './common/mat-editor.utils';
 })
 export class MatEditorComponent implements OnInit {
 
+  /**
+   * Sends the image url to the toolbar When it get sets through the directive
```
Comment (Anuranie) [inline projects/mat-editor/src/lib/mat-editor.component.ts]: Sends the image URL to the toolbar when it gets set to the directive

File: projects/mat-editor/src/lib/mat-editor.component.ts [hunk 8]
```diff
@@ -115,10 +122,11 @@ export class MatEditorComponent implements OnInit {
   ) { }
 
   ngOnInit() {
-
-    this.imageUploadService.onImageUpload.subscribe((file) => {
+    //subscribe for any images sent by the mat-editor toolbar component
```
Comment (Anuranie) [inline projects/mat-editor/src/lib/mat-editor.component.ts]: // Subscribe to any images sent by the Mat-Editor toolbar component

File: projects/demo/src/app/app.component.ts [hunk 9]
```diff
@@ -7,22 +7,38 @@ import { Component, OnInit } from '@angular/core';
 })
 export class AppComponent {
   title = 'Mat-Editor';
-  
+  count = 1; //count variable is used to demo the function uploadImage
```
Comment (Anuranie) [inline projects/demo/src/app/app.component.ts]: // Count variable is used to demo the function 'uploadImage'

**Review by Anuranie — APPROVED** [review]

**Pull request merged**